packages/legend-query-builder/src/stores/QueryBuilderTypeaheadHelper.ts

```diff
@@ -54,6 +69,57 @@ const initializeQueryBuilderState = (
   return builderState;
 };
 
+const applyMilestoningDates = (
+  propertyExpression: AbstractPropertyExpression,
+  observerContext: ObserverContext,
+): void => {
+  let currentExpression: ValueSpecification | undefined = propertyExpression;
+  while (currentExpression instanceof AbstractPropertyExpression) {
+    // check if the property is milestoned
+    if (
+      currentExpression.func.value.genericType.value.rawType instanceof Class &&
+      currentExpression.func.value._OWNER._generatedMilestonedProperties
+        .length !== 0 &&
+      currentExpression.func.value instanceof DerivedProperty &&
+      !currentExpression.func.value._OWNER.derivedProperties.includes(
+        currentExpression.func.value,
```

Comment:
can you explain when this last condition will be true?

Reply:
When the property is milestoned and it is actually a derived property.